### ISSUE TITLE
_import_rfc5915_der expects optional parameter to be present

### ISSUE THREAD

**Issue by johnboersi4** [Legrandin/pycryptodome#819]
I am working on a C++ / Python based security project where the C++ code generates a ECDSA private key that needs to be used by pycryptodome ECC for signing messages that are passed on to C++ code that has to validate the signature. The private key needs to be generated from C++ to keep all key generation in one place.

The PKCS#8 formatted ECDSA key complies nicely with the expected format for PKCS#8. But when DER decoding of the private key is done, it fails on incorrect number of elements, as _import_rfc5915_der expects the private key containing 3 or 4 parts. However, the private key octet stream generated from C++ code only contains 2 parts, namely version and the key itself.

Looking at the format specification, this is a valid combination as parameters and publicKey are optional, see below:

    # See RFC5915 https://tools.ietf.org/html/rfc5915
    #
    # ECPrivateKey ::= SEQUENCE {
    #           version        INTEGER { ecPrivkeyVer1(1) } (ecPrivkeyVer1),
    #           privateKey     OCTET STRING,
    #           parameters [0] ECParameters {{ NamedCurve }} OPTIONAL,
    #           publicKey  [1] BIT STRING OPTIONAL
    #    }

Hence, if I add the code change below, the C++ generated key is accepted (ECC.py):

```
def _import_rfc5915_der(encoded, passphrase, curve_oid=None):

    # See RFC5915 https://tools.ietf.org/html/rfc5915
    #
    # ECPrivateKey ::= SEQUENCE {
    #           version        INTEGER { ecPrivkeyVer1(1) } (ecPrivkeyVer1),
    #           privateKey     OCTET STRING,
    #           parameters [0] ECParameters {{ NamedCurve }} OPTIONAL,
    #           publicKey  [1] BIT STRING OPTIONAL
    #    }

    private_key = DerSequence().decode(encoded, nr_elements=(2, 3, 4))
    if private_key[0] != 1:
        raise ValueError("Incorrect ECC private key version")

    if len(private_key) > 2:
        try:
            parameters = DerObjectId(explicit=0).decode(private_key[2]).value
            if curve_oid is not None and parameters != curve_oid:
                raise ValueError("Curve mismatch")
            curve_oid = parameters
        except ValueError:
            pass
```

Changes made:
* added value 2 to nr_elements
* add check on length of private_key > 2

**Comment by Legrandin** [comment]
Fixed with 959c93a

**Comment by Legrandin** [comment]
Now in v3.21.0
